### PR TITLE
Changes due to gurumdds package integration

### PR DESCRIPTION
**Pull request by YoungJin-gurum**
# Package name:

rmw_gurumdds

# ROS Distro.

foxy

# The source is here:

https://github.com/ros2/rmw_gurumdds.git

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@YoungJin-gurum changes like this are usually handled via automatic pull request generation when creating new releases with `bloom-release`. Were you able to run bloom for these changes and if so, are there automated PRs that were opened to take care of these changes or was there an issue when running bloom?

**Comment by YoungJin-gurum** [comment]
@nuclearsandwich I know that pull requests are usually handled by generation automatically, but there was an issue when running bloom. 
When I tried bloom-release, I got the following error: Failed to open pull request: AssertionError - Duplicate package name 'gurumdds_cmake_module' exists in repository 'rosidl_typesupport_gurumdds' as well as in repository 'rmw_gurumdds'
So I manually edited it and uploaded a pull request. Please let me know if there is any other way I should take in this case. Thanks.

**Comment by methylDragon** [comment]
> @nuclearsandwich I know that pull requests are usually handled by generation automatically, but there was an issue when running bloom. When I tried bloom-release, I got the following error: Failed to open pull request: AssertionError - Duplicate package name 'gurumdds_cmake_module' exists in repository 'rosidl_typesupport_gurumdds' as well as in repository 'rmw_gurumdds' So I manually edited it and opened a pull request. Please let me know if there is any other way I should take in this case. Thanks.

Erm, I think that's because there are packages called `gurumdds_cmake_module` in both of those repos.

https://github.com/ros2/rmw_gurumdds/tree/master/gurumdds_cmake_module
https://github.com/ros2/rosidl_typesupport_gurumdds/tree/master/gurumdds_cmake_module

One of them will have to go, otherwise a user installing both will have conflicts.

**Comment by nuclearsandwich** [comment]
And just now there's a Foxy sync hold starting: https://discourse.ros.org/t/preparing-for-foxy-sync-2022-11-09/28209 with that in mind let's split the changes to Foxy in a separate set of PRs to do later.

**Comment by YoungJin-gurum** [comment]
@nuclearsandwich 

> 1. Update this PR so that it is just removing rosidl_typesupport_gurumdds

How do I update this PR?  Let me know what should I change.

File: foxy/distribution.yaml [hunk 1]
```diff
@@ -4992,11 +4992,11 @@ repositories:
     release:
       packages:
       - rmw_gurumdds_cpp
-      - rmw_gurumdds_shared_cpp
+      - gurumdds_cmake_module
       tags:
         release: release/foxy/{package}/{version}
       url: https://github.com/ros2-gbp/rmw_gurumdds-release.git
-      version: 1.2.0-1
+      version: 1.3.0-1
```
Comment (nuclearsandwich) [inline foxy/distribution.yaml:4999]: @YoungJin-gurum please revert these changes back to the 1.2.0-1 release and the original package list.

File: humble/distribution.yaml [hunk 2]
```diff
@@ -4251,11 +4251,11 @@ repositories:
     release:
       packages:
       - rmw_gurumdds_cpp
-      - rmw_gurumdds_shared_cpp
+      - gurumdds_cmake_module
       tags:
         release: release/humble/{package}/{version}
       url: https://github.com/ros2-gbp/rmw_gurumdds-release.git
-      version: 3.2.0-1
+      version: 3.3.0-1
```
Comment (nuclearsandwich) [inline humble/distribution.yaml:4258]: @YoungJin-gurum please revert these changes back to the 1.2.0-1 release and the original package list.

File: rolling/distribution.yaml [hunk 3]
```diff
@@ -4019,11 +4019,11 @@ repositories:
     release:
       packages:
       - rmw_gurumdds_cpp
-      - rmw_gurumdds_shared_cpp
+      - gurumdds_cmake_module
       tags:
         release: release/rolling/{package}/{version}
       url: https://github.com/ros2-gbp/rmw_gurumdds-release.git
-      version: 4.0.0-1
+      version: 4.1.1-1
```
Comment (nuclearsandwich) [inline rolling/distribution.yaml:4026]: @YoungJin-gurum please revert these changes back to the 1.2.0-1 release and the original package list.

File: foxy/distribution.yaml [hunk 4]
```diff
@@ -5713,23 +5713,6 @@ repositories:
...
-      - gurumdds_cmake_module
-      tags:
-        release: release/foxy/{package}/{version}
-      url: https://github.com/ros2-gbp/rosidl_typesupport_gurumdds-release.git
-      version: 1.1.1-1
-    source:
-      type: git
-      url: https://github.com/ros2/rosidl_typesupport_gurumdds.git
-      version: foxy
-    status: developed
```
Comment (nuclearsandwich) [inline foxy/distribution.yaml:5732]: Depending on the iteration process here, if we are otherwise ready to merge this while Foxy is still in a sync hold I'll split the changes for foxy out into a separate pull request before merging.

**Review by nuclearsandwich — CHANGES_REQUESTED** [review]

**Comment by nuclearsandwich** [comment]
> How do I update this PR? Let me know what should I change.

I've added some comments to each section. Let me know if I can provide further explanation.

**Comment by YoungJin-gurum** [comment]
@nuclearsandwich I revert the package version and list, just removing rosidl_typesupport_gurumdds.
Please let me know if there are any additional corrections I need to make.

**Comment by ivanpauno** [comment]
[Holding for Humble sync](https://discourse.ros.org/t/preparing-for-humble-sync-and-patch-release-2022-11-21/28234)

**Comment by ivanpauno** [comment]
[Holding for Foxy sync](https://discourse.ros.org/t/preparing-for-foxy-sync-2022-11-09/28209)

**Review by ivanpauno — APPROVED** [review]
LGTM!
The `rolling` part can be merged, but we need to wait because of the `foxy`/`humble` syncs

**Comment by nuclearsandwich** [comment]
@YoungJin-gurum I'm also starting a Rolling sync this week as well.([link](https://discourse.ros.org/t/preparing-for-rolling-sync-2022-11-17/28330)). If Rolling merges first, I'll refactor this PR so that we can update Rolling immediately after the sync.

Thanks for iterating on this so far, we'll take care of the changes as each release completes.

**Review by nuclearsandwich — APPROVED** [review]
All three of these distros are out of sync hold.

**Comment by nuclearsandwich** [comment]
@YoungJin-gurum you should now be able to bloom the new versions of rmw_gurumdds normally. Thanks again for your patience during this process.
